### ISSUE TITLE
[Bug]: Masking does not work on text2image controlnets

### ISSUE THREAD

**Issue by Aiboi-png** [Panchovix/stable-diffusion-webui-reForge#342]
### Checklist

- [x] The issue exists after disabling all extensions
- [x] The issue exists on a clean installation of webui
- [ ] The issue is caused by an extension, but I believe it is caused by a bug in the webui
- [x] The issue exists in the current version of the webui
- [x] The issue has not been reported before recently
- [ ] The issue has been reported before but has not been fixed yet

### What happened?

When masking a controlnet image the resulting image doesn't respect the mask.

### Steps to reproduce the problem

![Image](https://github.com/user-attachments/assets/bd2417a6-e7a6-4cba-9359-0f8bf49da92e)

Load the supplied image into controlnet
Select the controlnet type you wish to use. I have tested this using "canny" and "depth"
Lock the seed and generate an image
After generating the image, mask out the square
Generate a new image
The resulting image should be the same. 

### What should have happened?

The masked area of the image should/should not be effected by the controlnet

### What browsers do you use to access the UI ?

Google Chrome

### Sysinfo

[sysinfo-2025-03-25-04-44.json](https://github.com/user-attachments/files/19523638/sysinfo-2025-03-25-04-44.json)

### Console logs

```Shell
Python 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)]
Version: f1.7.0-v1.10.1RC-latest-2165-g25dcc6be
Commit hash: 25dcc6be646a2e973b6566c1bd149a74ecf09f44
E:\Data\Packages\A1\extensions-builtin\forge_legacy_preprocessors\install.py:2: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  import pkg_resources
E:\Data\Packages\A1\extensions-builtin\sd_forge_controlnet\install.py:2: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  import pkg_resources
Launching Web UI with arguments: --gradio-allowed-path 'E:\Data\Images'
Total VRAM 24564 MB, total RAM 130262 MB
pytorch version: 2.3.1+cu121
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 4090 : native
Hint: your device supports --pin-shared-memory for speed improvements if having 16GB or more VRAM. It may cause issues with 12GB VRAM or less.
Hint: your device supports --cuda-malloc for a bit less vram usage and small speed improvement
Hint: your device supports --cuda-stream for potential speed improvements but it may cause issues on some GPUs that have 6-8GB VRAM or less.
E:\Data\Packages\A1\modules_forge\initialization.py:55: UserWarning: expandable_segments not supported on this platform (Triggered internally at ..\c10/cuda/CUDAAllocatorConfig.h:28.)
  torch.zeros((1, 1)).to(device, torch.float32)
CUDA Stream Activated:  False
Using pytorch attention
ControlNet preprocessor location: E:\Data\Packages\A1\models\ControlNetPreprocessor
Memory change: 1.45 MB (0.86 GB total)
Loading model sd\Real & Semi\dreamshaper_8.safetensors [879db523c3] (1 of 1)
Loading weights [879db523c3] from E:\Data\Packages\A1\models\Stable-diffusion\sd\Real & Semi\dreamshaper_8.safetensors
2025-03-29 18:17:36,335 - ControlNet - INFO - ControlNet UI callback registered.
WARNING:root:clip missing: ['clip_l.text_projection', 'clip_l.logit_scale']
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Reloading VAE
Startup time: 15.9s (prepare environment: 2.4s, import torch: 4.6s, import gradio: 0.6s, setup paths: 0.6s, initialize shared: 0.3s, other imports: 0.3s, load scripts: 4.3s, create ui: 1.7s, gradio launch: 0.9s).
Model sd\Real & Semi\dreamshaper_8.safetensors [879db523c3] loaded in 2.6s (memory cleanup: 0.5s, load weights from disk: 0.1s, forge load real models: 1.5s, load VAE: 0.2s, calculate empty prompt: 0.3s).
Memory change: 2.22 MB (3.63 GB total)
2025-03-29 18:17:53,141 - ControlNet - INFO - ControlNet Input Mode: InputMode.SIMPLE
2025-03-29 18:17:53,145 - ControlNet - INFO - Using preprocessor: depth_midas
2025-03-29 18:17:53,145 - ControlNet - INFO - preprocessor resolution = 512
Automatic Memory Management: 659 Modules in 0.10 seconds.
[] []
2025-03-29 18:17:55,751 - ControlNet - INFO - Current ControlNet ControlNetPatcher: E:\Data\Packages\A1\models\ControlNet\ControlNet\control_v11f1p_sd15_depth.pth
2025-03-29 18:17:55,792 - ControlNet - INFO - ControlNet Method depth_midas patched.
100%|██████████| 20/20 [00:04<00:00,  4.90it/s]
Total progress: 100%|██████████| 20/20 [00:04<00:00,  4.31it/s]s]
```

### Additional information

I have tested this with multiple types of models, 1.5, XL, lightning, etc. The issue persists. I have also used differing sampler/schedulers, and the issue persists. A mask does appear to be generated, and shows up after the image is made. The issue occurs both when drawing a mask straight onto the controlnet image, and when using the option to have an independent mask.
